### DISCUSSION TITLE
no display?

### DISCUSSION THREAD

**Discussion by xdewx** [rustdesk/rustdesk#1615] [Q&A]
My remote host doesn't have a display, so I want to use rustdesk to connect to it. But when I start rustdesk locally and remotely, the local client prompts for no display error. How should this be solved?

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/418#issuecomment-1233018617

**Comment by Osiware** [comment]
(debian 11 & debian 10)
In my case I solved it by installing and modify gdm3 configuration whit this procedure :

1º Change GDM3 Configuration ;
Change : /etc/gdm3/daemon.conf
Set :
#WaylandEnable=false
to ;
WaylandEnable=false

2º install  no monitor (dummy) :
sudo apt-get install xserver-xorg-video-dummy

And creating or changing in the X11 configuration:

Then write it in the /usr/share/X11/xorg.conf.d/xorg.conf (or possibly /etc/X11/xorg.conf) file (create one, if it does not exist):

#########################################

Section "Device"
    Identifier  "Configured Video Device"
    Driver      "dummy"
EndSection

Section "Monitor"
    Identifier  "Configured Monitor"
    HorizSync 31.5-48.5
    VertRefresh 50-70
EndSection

Section "Screen"
    Identifier  "Default Screen"
    Monitor     "Configured Monitor"
    Device      "Configured Video Device"
    DefaultDepth 24
    SubSection "Display"
    Depth 24
    Modes "1024x800"
    EndSubSection
EndSection

########################################·

REBOOT 

(the client computer has not a monitor in the case of putting one, nothing will be seen until we change this configuration via, for example, a remote connection with SSH or Putty, ....) since we have told X11 to use a monitor non-existent. So I advise you to keep in mind the changes

**Reply by kitmanten** [reply]
it does fit on ubuntu 22.04?

**Reply by CappyBeara** [reply]
Yes, this worked in ubuntu 22.04

I had to create the xorg.conf

sudo nano /usr/share/X11/xorg.conf.d/xorg.conf 

(paste config)

ctrl + x 
y
enter

reboot

**Reply by platinaCoder** [reply]
Thank you so much, it works just with less resolution, gonna play some more with these settings later.

**Reply by eliaspfeffer** [reply]
> Thank you so much, it works just with less resolution, gonna play some more with these settings later.

Did you  find a solution for higher resolution?

**Reply by JIANGTUNAN** [reply]
> > Thank you so much, it works just with less resolution, gonna play some more with these settings later.
> 
> Did you find a solution for higher resolution?


```
Section "Device"
    Identifier  "Configured Video Device"
    Driver      "dummy"
    VideoRam 256000
EndSection

Section "Monitor"
    Identifier  "Configured Monitor"
    HorizSync 5.0 - 1000.0
    VertRefresh 5.0 - 200.0
    ModeLine "1920x1080" 148.50 1920 2448 2492 2640 1080 1084 1089 1125 +Hsync +Vsync
EndSection

Section "Screen"
    Identifier  "Default Screen"
    Monitor     "Configured Monitor"
    Device      "Configured Video Device"
    DefaultDepth 24
    SubSection "Display"
    Depth 24
    Modes "1920x1080" "1280x800" "1024x768" "1920x1080" "1440x900"
    EndSubSection
EndSection
```


[refer to](https://askubuntu.com/questions/1396903/dummy-monitor-in-21-10)

**Reply by artyPD** [reply]
> > > Thank you so much, it works just with less resolution, gonna play some more with these settings later.
> > 
> > 
> > Did you find a solution for higher resolution?
> 
> ```
> Section "Device"
>     Identifier  "Configured Video Device"
>     Driver      "dummy"
>     VideoRam 256000
> EndSection
> 
> Section "Monitor"
>     Identifier  "Configured Monitor"
>     HorizSync 5.0 - 1000.0
>     VertRefresh 5.0 - 200.0
>     ModeLine "1920x1080" 148.50 1920 2448 2492 2640 1080 1084 1089 1125 +Hsync +Vsync
> EndSection
> 
> Section "Screen"
>     Identifier  "Default Screen"
>     Monitor     "Configured Monitor"
>     Device      "Configured Video Device"
>     DefaultDepth 24
>     SubSection "Display"
>     Depth 24
>     Modes "1920x1080" "1280x800" "1024x768" "1920x1080" "1440x900"
>     EndSubSection
> EndSection
> ```
> 
> [refer to](https://askubuntu.com/questions/1396903/dummy-monitor-in-21-10)

While this worked for me (thanks). It now doesn't show a GUI over the HDMI ports. Instead it simply shows a flashing cursor in the top left. I can still see the GUI over rustdesk just fine.

Is it possible to still have the GUI seen over the HDMI ports as well as over rustdesk?

**Comment by guglovich** [comment]
Any advice for those who have a monitor but the same error?

**Comment by khdoelger** [comment]
@xdewx @kitmanten a "HDMI Display Emulator" did the trick for me for the "no display" problem. 
I bought this one for my displayless mini pc and it worked (ubuntu 22.04): https://www.amazon.de/gp/product/B0788NPQFV?th=1

**Comment by vincentpaul520** [comment]
Debian12.5  KDE X11  :

```
sudo apt-get install xserver-xorg-video-dummy

sudo nano /etc/X11/xorg.conf

Section "Device"
    Identifier "dummy"
    Driver "dummy"
    Option "DummyDevice" "true"
    Option "VideoRam" "256000"
EndSection

Section "Monitor"
    Identifier "dummy"
    HorizSync 5.0 - 1000.0
    VertRefresh 5.0 - 200.0
    Modeline "1920x1080i"x0.0 74.25 1920 2008 2052 2200 1080 1084 1094 1125 interlace +hsync +vsync (33.8 kHz e)
    Modeline "1920x1080i"x0.0 74.25 1920 2448 2492 2640 1080 1084 1094 1125 interlace +hsync +vsync (28.1 kHz e)
EndSection

Section "Screen"
    Identifier "dummy"
    Device "dummy"
    Monitor "dummy"
    DefaultDepth 24
    SubSection "Display"
        Depth 24
        Modes "1920x1080i"
    EndSubSection
EndSection
```
【Manual】

After plugging in the hardware monitor again, it will not display, so it needs to be disabled
Temporarily cancel virtual display and restart

Sudo mv/etc/X11/xorg. conf/etc/X11/xorg. conf. backup
reboot


Modify back to virtual display

Sudo mv/etc/X11/xorg. conf. backup/etc/X11/xorg. conf

【automatic】
[Automatic Script Processing - Virtual and Physical Automatic Switching]
Xrandr -- listmonitors

Monitoring: 2
0:+* HDMI-1 1920/480x1080/270+0+0 HDMI-1
1: +HDMI-2 1920/480x1080/270+0+0 HDMI-2


详细步骤
步骤1：备份默认配置（如果尚未备份）

bash
复制
sudo cp /etc/X11/xorg.conf /etc/X11/xorg.conf.backup
步骤2：创建虚拟显示器配置文件
在 /etc/X11/xorg.conf.d/ 目录下创建一个名为 99-virtual-display.conf 的文件，并使用 sudo 权限编辑它：

bash
复制
sudo nano /etc/X11/xorg.conf.d/99-virtual-display.conf
在编辑器中输入以下内容作为虚拟显示器的配置：

apache
复制
Section "Device"
    Identifier "dummy"
    Driver "dummy"
    Option "DummyDevice" "true"
EndSection

Section "Monitor"
    Identifier "dummy"
    HorizSync 5.0 - 1000.0
    VertRefresh 5.0 - 200.0
    Modeline "800x600" 24.0 800 832 928 1024 600 604 608 624
EndSection

Section "Screen"
    Identifier "dummy"
    Device "dummy"
    Monitor "dummy"
    DefaultDepth 24
    SubSection "Display"
        Depth 24
        Modes "800x600"
    EndSubSection
EndSection
步骤3：创建udev规则
在 /etc/udev/rules.d/ 目录下创建一个名为 99-monitor-hotplug.rules 的文件，并使用 sudo 权限编辑它：

bash
复制
sudo nano /etc/udev/rules.d/99-monitor-hotplug.rules
在编辑器中输入以下内容作为udev规则：

复制
ACTION=="change", SUBSYSTEM=="drm", ENV{HOTPLUG}=="1", RUN+="/usr/bin/systemctl start monitor-config.service"
步骤4：创建systemd服务
在 /etc/systemd/system/ 目录下创建一个名为 monitor-config.service 的文件，并使用 sudo 权限编辑它：

bash
复制
sudo nano /etc/systemd/system/monitor-config.service
在编辑器中输入以下内容作为systemd服务：

ini
复制
[Unit]
Description=Configure monitors based on presence
Requires=display-manager.service
After=display-manager.service

[Service]
Type=simple
ExecStart=/usr/local/bin/configure-monitors.sh
Restart=no

[Install]
WantedBy=multi-user.target
步骤5：编写监控显示器状态的脚本
创建一个名为 /usr/local/bin/configure-monitors.sh 的脚本文件，并使用 sudo 权限编辑它：

bash
复制
sudo nano /usr/local/bin/configure-monitors.sh
将以下内容复制到脚本文件中：

bash
复制
#!/bin/bash

HAS_PHYSICAL_DISPLAY=$(xrandr --listmonitors | grep 'connected' | wc -l)

if [ "$HAS_PHYSICAL_DISPLAY" -eq 0 ]; then
    sudo ln -sf /etc/X11/xorg.conf.d/99-virtual-display.conf /etc/X11/xorg.conf
else
    sudo rm -f /etc/X11/xorg.conf
fi

sudo systemctl restart display-manager

exit 0
步骤6：赋予脚本执行权限

bash
复制
sudo chmod +x /usr/local/bin/configure-monitors.sh
步骤7：启用并启动systemd服务

bash
复制
sudo systemctl daemon-reload
sudo systemctl enable monitor-config.service
sudo systemctl start monitor-config.service

**Comment by ggibby** [comment]
Three years later this is still a bug. 
No problem connecting to two headless Windows Server boxes (though the resolution is garbage).
RD refuses to connect to a Windows 10 desktop, but RDP via RustDesk works perfectly.
Completely removed and reinstalled on Windows, working on that machine.
The Linux server host keeps losing its BIOS settings and after a reboot errors out with the no display error.

**Reply by eliaspfeffer** [reply]
Not sure if this solves your problem but I've connected my linux machine to a monitor, switched it over to a virtual monitor (not sure anymore what command I've used), plugged out the real monitor and connected to the machine via rustdesk. Now I see the virtual monitor without a real monitor being connected to it.
